server/homebrew.api.js

```diff
@@ -89,148 +135,176 @@ const newBrew = async (req, res)=>{
 };
 
+const deleteGoogleBrew = async (account, id, editId, res)=>{
+	const auth = await GoogleActions.authCheck(account, res);
+	await GoogleActions.deleteGoogleBrew(auth, id, editId);
+	return true;
+};
+
 const deleteBrew = async (req, res)=>{
-	if(req.params.id.length > 12) {
-		const deleted = await deleteGoogleBrew(req.account, req.params.id, res)
+	const { brew, account } = req;
+	if(brew.googleId) {
+		const googleDeleted = await deleteGoogleBrew(account, brew.googleId, brew.editId, res)
 			.catch((err)=>{
 				res.status(500).send(err);
 			});
-		if(deleted) return res.status(200).send();
-	} else {
-		const deleted = await deleteLocalBrew(req.account, req.params.id)
-			.catch((err)=>{
-				res.status(err.status).send(err.message);
-			});
-		if(deleted) return res.status(200).send(deleted);
-		return res.status(200).send();
-	}
-};
-
-const deleteLocalBrew = async (account, id)=>{
-	const brew = await HomebrewModel.findOne({ editId: id });
-	if(!brew) {
-		throw { status: 404, message: 'Can not find homebrew with that id' };
+		if(!googleDeleted) {
+			if(!res.headersSent) {
+				res.status(500).send('Unable to delete brew from Google');
+			}
+			return;
+		}
 	}
 
-	if(account) {
-		// Remove current user as author
-		brew.authors = _.pull(brew.authors, account.username);
-		brew.markModified('authors');
-	}
+	if(brew._id) {
+		if(account) {
+			// Remove current user as author
+			brew.authors = _.pull(brew.authors, account.username);
+			brew.markModified('authors');
+		}
 
-	if(brew.authors.length === 0) {
-		// Delete brew if there are no authors left
-		await brew.remove()
-			.catch((err)=>{
-				console.error(err);
-				throw { status: 500, message: 'Error while removing' };
-			});
-	} else {
-		// Otherwise, save the brew with updated author list
-		return await brew.save()
-			.catch((err)=>{
-				throw { status: 500, message: err };
-			});
+		if(brew.authors.length === 0) {
+			// Delete brew if there are no authors left
+			await brew.remove()
+				.catch((err)=>{
+					console.error(err);
+					throw { status: 500, message: 'Error while removing' };
+				});
+		} else {
+			// Otherwise, save the brew with updated author list
+			await brew.save()
+				.catch((err)=>{
+					throw { status: 500, message: err };
+				});
+		}
 	}
-};
 
-const deleteGoogleBrew = async (account, id, res)=>{
-	const auth = await GoogleActions.authCheck(account, res);
-	await GoogleActions.deleteGoogleBrew(auth, id);
-	return true;
+	return res.status(204).send();
 };
 
 router.post('/api', asyncHandler(newBrew));
-router.put('/api/:id', asyncHandler(updateBrew));
-router.put('/api/update/:id', asyncHandler(updateBrew));
-router.delete('/api/:id', asyncHandler(deleteBrew));
-router.get('/api/remove/:id', asyncHandler(deleteBrew));
-
-module.exports = router;
+router.put('/api/:id', asyncHandler(getBrew('edit')), asyncHandler(updateBrew));
+router.put('/api/update/:id', asyncHandler(getBrew('edit')), asyncHandler(updateBrew));
```

Comment:
Help me understand why we need to call `getBrew` on all of these api calls? Isn't the brew already provided from the client in `req` by the time we get to this point? Unless there's something changed in this PR resulting in cases requiring re-fetching brews before we handle them?

Reply:
It is, but in all of these we have to fetch the instance of the brew from the database in order to properly save it, since save is a method on the Mongoose object. Since all of them involve stubbing now, that has to be done in all routes here.

Reply:
Right, we need the mongo object. Gotcha. Ok.